src/document.ts

```diff
@@ -95,7 +96,11 @@ export class Document {
         static: this.pod.staticFile.bind(this.pod),
       },
     };
-    return this.renderer.render(this.view, context);
+    const renderer = this.renderer;
+    if (!renderer) {
+      throw Error(`No renderer found, can't render: ${this}`);
+    }
+    return (this.renderer as Renderer).render(this.view, context);
```

Comment:
maybe use your local "renderer" value here so you don't have to type cast using `as`.